### PR TITLE
New Helper Functions

### PR DESCRIPTION
**Pull request by Goncalo-Marques**
Added two new helper functions.

mathf:
- Approximately `compares two floating point values and returns true if they are similar`

vector2:
-  Zero `returns a vector with a value of 0 in both fields, X and Y. Provides a shorthand for Vector2{X: 0, Y: 0}`

### CODE REVIEW

File: mathf/mathf.go [hunk 1]
```diff
@@ -115,3 +115,9 @@ func QuadraticFormula(a, b, c float64) (x1, x2 float64) {
 	x2 = (-b - discriminant) / divisor
 	return
 }
+
+// Approximately compares two floating point values and returns true if they are similar
+func Approximately(a, b float64) bool {
+	v := a - b
+	return v >= -Epsilon64 && v <= Epsilon64
```
Comment (EmanuelFeij) [inline mathf/mathf.go:122]: Maybe `return math.abs(a-b) <= Eplison64` is enough

**Review by EmanuelFeij — APPROVED** [review]

File: mathf/mathf.go (hunk 1 above)
Comment (Goncalo-Marques) [reply mathf/mathf.go:122]: Doesn't `math.Abs()` use more processing than this approach?

Comment (EmanuelFeij) [reply mathf/mathf.go:122]: Seems that is some bitwise operations, get pointer to the value  and flipping, if needed, the signal bit. 
`func Abs(x float64) float64 {
	return Float64frombits(Float64bits(x) &^ (1 << 63))
}
`

Comment (diogojsdias) [reply mathf/mathf.go]: The only way to check this is by benchmarking the code:

```
func BenchmarkApproximatelyWithAbs(b *testing.B) {
	f1 := float64(1.00)
	f2 := float64(2*(1.0 * 0.5))
	for i := 0; i < b.N; i++ {
		ApproximatelyWithAbs(f1, f2)
	}
}

func BenchmarkApproximatelyWithSubtraction(b *testing.B) {
	f1 := float64(1.00)
	f2 := float64(2*(1.0 * 0.5))
	for i := 0; i < b.N; i++ {
		ApproximatelyWithSubtraction(f1, f2)
	}
}

func ApproximatelyWithAbs(a, b float64) bool {
	return math.Abs(a-b) <= Epsilon64
}

func ApproximatelyWithSubtraction(a, b float64) bool {
	v := a - b
	return v >= -Epsilon64 && v <= Epsilon64
}
```

And from the results, it seems there's a very minor advantage on the `ApproximatelyWithSubtraction` implementation:

``` 
cpu: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
BenchmarkApproximatelyWithSubtraction
BenchmarkApproximatelyWithSubtraction-8   	1000000000	         0.3070 ns/op
```

``` 
cpu: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
BenchmarkApproximatelyWithAbs
BenchmarkApproximatelyWithAbs-8   	1000000000	         0.3635 ns/op
```

**Review by diogojsdias — APPROVED** [review]
:ok_hand:

**Pull request merged**